unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -6936,129 +6936,148 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
-                            transform.position = p;
+            // iterate over each reasonable agent pose
+            foreach (float horizon in horizons) {
+                // recall that horizon=60 is look down 60 degrees and horizon=-30 is look up 30 degrees
+                if (horizon > maxDownwardLookAngle || horizon < -maxUpwardLookAngle) {
+                    errorMessage = $"Each horizon must be in [{-maxUpwardLookAngle}:{maxDownwardLookAngle}]";
+                    actionFinished(success: false);
+                }
+                m_Camera.transform.localEulerAngles = new Vector3(horizon, 0f, 0f);
+
+                foreach (bool standing in standings) {
+                    if (standing) stand(); else crouch();
 
+                    foreach (float rotation in rotations) {
+                        transform.rotation = Quaternion.Euler(new Vector3(0.0f, rotation, 0.0f));
+
+                        foreach (Vector3 position in filteredPositions) {
+                            transform.position = position;
                             if (objectIsCurrentlyVisible(theObject, maxVisibleDistance)) {
-                                goodLocationsDict["x"].Add(p.x);
-                                goodLocationsDict["y"].Add(p.y);
-                                goodLocationsDict["z"].Add(p.z);
-                                goodLocationsDict["rotation"].Add(90.0f * i);
-                                goodLocationsDict["standing"].Add((1 - j) * 1.0f);
-                                goodLocationsDict["horizon"].Add(m_Camera.transform.localEulerAngles.x);
+                                validAgentPoses["x"].Add(position.x);
+                                validAgentPoses["y"].Add(position.y);
+                                validAgentPoses["z"].Add(position.z);
+                                validAgentPoses["rotation"].Add(rotation);
+                                validAgentPoses["standing"].Add(standing);
+                                validAgentPoses["horizon"].Add(horizon);
 
-#if UNITY_EDITOR
-                                // In the editor, draw lines indicating from where the object was visible.
-                                Debug.DrawLine(p, p + transform.forward * (gridSize * 0.5f), Color.red, 20f);
-#endif
+                                #if UNITY_EDITOR
+                                    // In the editor, draw lines indicating from where the object was visible.
+                                    Debug.DrawLine(position, position + transform.forward * (gridSize * 0.5f), Color.red, 20f);
+                                #endif
                             }
                         }
                     }
                 }
             }
 
-            if (wasStanding) {
-                stand();
-            } else {
-                crouch();
-            }
+            // restore old agent pose
+            if (wasStanding) stand(); else crouch();
```

Comment:
Again here. I understand the desire to save space but I consider these bad style as it's very easy to break things without clear code blocks.